spec/system/analysis_spec.rb

```diff
@@ -5,8 +5,46 @@
   let(:school_name) { 'Theresa Green Infants'}
   let(:school)     { create(:school, name: school_name)}
 
-  it 'I can visit the school analysis page' do
+  it 'redirects back to the school home page if no meters are set' do
     visit school_analysis_path(school)
-    expect(page.has_content? school_name).to be true
+    expect(page).to have_content 'Analysis is currently unavailable due to a lack of validated meter readings'
+  end
+
+  context 'when a school has an electricity meter' do
+    let!(:meter) { create(:electricity_meter_with_validated_reading, name: 'Electricity meter', school: school) }
+
+    it 'requests for floor area and pupil numbers to be populated' do
+      stub_out_the_aggregation_etc
+      visit school_analysis_path(school)
+      expect(page).to have_content('Please edit the school details')
+    end
+
+    it 'does not request for floor area and pupil numbers to be populated if they are' do
+      floor_area = 20
+      number_of_pupils = 100
+      school.update(floor_area: floor_area, number_of_pupils: number_of_pupils)
+      stub_out_the_aggregation_etc
+      visit school_analysis_path(school)
+      expect(page).to_not have_content('Please edit the school details')
+      expect(page).to have_content(floor_area)
+      expect(page).to have_content(number_of_pupils)
+    end
+
+    it 'allows units to be selected', js: :true do
+      stub_out_the_aggregation_etc
+      visit school_analysis_path(school)
+      expect(page).to have_content('Currently your measurements are in energy used in kilowatt-hours')
+      click_on('Change energy usage units', match: :first)
+      Measurements::MEASUREMENT_OPTIONS.values.each { |energy_description| expect(page.has_content?(energy_description.capitalize)).to be true }
+      expect(find('#measurement_kwh').selected?).to be true
+      expect(find('#measurement_co2').selected?).to be false
+    end
+  end
+
+  def stub_out_the_aggregation_etc
```

Comment:
Pretty yucky that we have to do this - shows that my implementation of the analysis stuff needs to know too much!